openedx/core/djangoapps/course_groups/views.py

```diff
@@ -379,3 +388,226 @@ def debug_cohort_mgmt(request, course_key_string):
+        if not assignment_type:
+            raise self.api_error(status.HTTP_400_BAD_REQUEST,
+                                 'Assignment type must be specified.',
+                                 'missing-assignment-type')
+        return Response(_get_cohort_representation(cohorts.add_cohort(course_key, name, assignment_type), course))
+
+    def patch(self, request, course_key_string, cohort_id=None):
+        """
+        Endpoint to update a cohort information, including:
+        """
+        name = request.data.get('name')
+        assignment_type = request.data.get('assignment_type')
+        if None in {cohort_id, name, assignment_type}:
+            raise self.api_error(status.HTTP_400_BAD_REQUEST,
+                                 'Request must include cohort id, name and assignment type.',
+                                 'missing-field')
+        course_key, _ = api.get_course(request, course_key_string)
+        cohort = cohorts.get_cohort_by_id(course_key, cohort_id)
+        if name != cohort.name:
+            if cohorts.is_cohort_exists(course_key, name):
+                raise self.api_error(status.HTTP_400_BAD_REQUEST,
+                                     'A cohort with the same name already exists.',
+                                     'cohort-name-exists')
+            cohort.name = name
+            cohort.save()
+        cohorts.set_assignment_type(cohort, assignment_type)
+        return Response(status=status.HTTP_204_NO_CONTENT)
+
+
+class CohortUsers(DeveloperErrorViewMixin, APIPermissions):
+    """
+    **Use Cases**
+
+        Removes an user from a cohort.
+        Add a user to a specific cohort.
+
+    **Example Requests**:
+
+        DELETE /api/cohorts/v1/courses/{course_id}/cohorts/{cohort_id}/users/{username}
+        POST /api/cohorts/v1/courses/{course_id}/cohorts/{cohort_id}/users/{username}
+        POST /api/cohorts/v1/courses/{course_id}/cohorts/{cohort_id}/users
+
+    **Response Values**
+        * success: Boolean indicating if the operation was successful.
+        * added: Usernames/emails of the users that have been added to the cohort.
+        * changed: Usernames/emails of the users that have been moved to the cohort.
+        * present: Usernames/emails of the users already present in the cohort.
+        * unknown: Usernames/emails of the users with an unknown cohort.
+        * preassigned: Usernames/emails of unenrolled users that have been preassigned to the cohort.
+        * invalid: Invalid emails submitted.
+    """
+
+    # pylint: disable=unused-argument
+    def delete(self, request, course_key_string, cohort_id, username=None):
+        """
+        Removes and user from a specific cohort.
+
+        Note: It's better to use the post method to move users between cohorts.
+        """
+        course_key, _ = api.get_course(request, course_key_string)
+        try:
+            api.remove_user_from_cohort(course_key, username)
```

Comment:
nit: perhaps make username=None a method not allowed error.

Reply:
Also no longer checking for CohortMemborship.DoesNotExist

Reply:
As before, the check is handled [here](https://github.com/edx/edx-platform/pull/18975/files/15a863e1ad530673579684e718d4c9438a8e5429#diff-7573f15896246dec87f062d444947cc4R40).

👍 for the nit